app/controllers/async_info_controller.rb

```diff
@@ -52,27 +52,33 @@ def broadcast_data
 
   def user_data
     Rails.cache.fetch(user_cache_key, expires_in: 15.minutes) do
-      {
-        id: @user.id,
-        name: @user.name,
-        username: @user.username,
-        profile_image_90: ProfileImage.new(@user).get(width: 90),
-        followed_tags: @user.cached_followed_tags.to_json(only: %i[id name bg_color_hex text_color_hex hotness_score], methods: [:points]),
-        followed_user_ids: @user.cached_following_users_ids,
-        followed_podcast_ids: @user.cached_following_podcasts_ids,
-        reading_list_ids: ReadingList.new(@user).cached_ids_of_articles,
-        blocked_user_ids: @user.all_blocking.pluck(:blocked_id),
-        saw_onboarding: @user.saw_onboarding,
-        checked_code_of_conduct: @user.checked_code_of_conduct,
-        checked_terms_and_conditions: @user.checked_terms_and_conditions,
-        display_sponsors: @user.display_sponsors,
-        display_announcements: @user.display_announcements,
-        trusted: @user.trusted,
-        moderator_for_tags: @user.moderator_for_tags,
-        config_body_class: @user.config_body_class,
-        pro: @user.pro?,
-        created_at: @user.created_at
-      }
+      data =
+        {
+          blocked_user_ids: @user.all_blocking.pluck(:blocked_id),
+          checked_code_of_conduct: @user.checked_code_of_conduct,
+          checked_terms_and_conditions: @user.checked_terms_and_conditions,
+          config_body_class: @user.config_body_class,
+          created_at: @user.created_at,
+          display_announcements: @user.display_announcements,
+          display_sponsors: @user.display_sponsors,
+          followed_podcast_ids: @user.cached_following_podcasts_ids,
+          followed_tags: @user.cached_followed_tags.to_json(only: %i[id name bg_color_hex text_color_hex hotness_score], methods: [:points]),
+          followed_user_ids: @user.cached_following_users_ids,
+          id: @user.id,
+          moderator_for_tags: @user.moderator_for_tags,
+          name: @user.name,
+          pro: @user.pro?,
+          profile_image_90: ProfileImage.new(@user).get(width: 90),
+          reading_list_ids: ReadingList.new(@user).cached_ids_of_articles,
+          saw_onboarding: @user.saw_onboarding,
+          trusted: @user.trusted,
+          username: @user.username
+        }
+
+      # TODO: (Alex Smith) remove this when ready
+      data[:email] = @user.email if @user.has_role?(:super_admin) || @user.has_role?(:admin)
```

Comment:
Let's just add `:email` to the hash right now with the above comment next to it rather than doing a role lookup here. That's going to add 2 more db hits which I would rather avoid in exchange for a couple more bits in the hash.

Reply:
That's a great point, good call. I'll have to scrap `email` altogether 😢  - totally doable, just causes me much more pain in the next few PRs.

It's not ready to be applied to all users yet and I don't think it's worth the trouble of me caching the roles to make this more performant when the check is going to be removed anyway.